### PR TITLE
karmadactl get uses factory to access cluster

### PR DESCRIPTION
**Pull request by helen-frank**
**What type of PR is this?**
/kind feature

**Which issue(s) this PR fixes**:
Fixes #
Part of [#2349](https://github.com/karmada-io/karmada/issues/2349)
**Special notes for your reviewer**:
Test
```shell
go run ./cmd/karmadactl/karmadactl.go -kubeconfig /etc/karmada/karmada-apiserver.config  -n test1 -C kind-multi-node1 get po                                                                         
NAME                      CLUSTER            READY   STATUS    RESTARTS   AGE
nginx1-56db585f94-lsv8b   kind-multi-node1   1/1     Running   0          5d13h
```
**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

**Comment by carlory** [comment]
sorry for too late reply, I will review it tomorrow.

**Comment by carlory** [comment]
please use the [FactoryForMemberCluster]( https://github.com/karmada-io/karmada/blob/master/pkg/karmadactl/util/factory.go#L25) function to access to member cluster

FYI:  https://github.com/karmada-io/karmada/blob/master/pkg/karmadactl/exec.go#L103

File: pkg/karmadactl/get.go [hunk 1]
```diff
@@ -916,81 +874,6 @@ func (g *CommandGetOptions) transformRequests(req *rest.Request) {
 	}, ","))
 }
 
-func (g *CommandGetOptions) getRBInKarmada(namespace string, config *rest.Config) error {
```
Comment (lonelyCZ) [inline pkg/karmadactl/get.go:919]: We shouldn't delete it that is designed for showing `ADOPTION` column.

**Review by lonelyCZ — COMMENTED** [review]
Thanks for your hard work. @helen-frank

**Comment by lonelyCZ** [comment]
There seem some problems.

```
[root@master67 karmada]# ./karmadactl get pod -A --kubeconfig /etc/karmada/karmada-apiserver.config
No resources found
```

**Comment by RainbowMango** [comment]
I can't retrigger the test, don't know why. Maybe you can rebase your code and push again.

File: pkg/karmadactl/get.go [hunk 2]
```diff
@@ -285,19 +280,17 @@ func (g *CommandGetOptions) Run(karmadaConfig KarmadaConfig, cmd *cobra.Command,
 	clusterInfos := make(map[string]*ClusterInfo)
 	RBInfo = make(map[string]*OtherPrint)
 
-	karmadaRestConfig, err := clusterInfoInit(g, karmadaConfig, clusterInfos)
-	if err != nil {
+	if err := clusterInfoInit(g, f, clusterInfos); err != nil {
```
Comment (lonelyCZ) [inline pkg/karmadactl/get.go:283]: It seems unnecessary to define `clusterInfos` in `Run` function.

File: pkg/karmadactl/get.go [hunk 3]
```diff
@@ -908,6 +879,28 @@ func getFactory(clusterName string, clusterInfos map[string]*ClusterInfo, namesp
 	return cmdutil.NewFactory(matchVersionKubeConfigFlags)
 }
 
+func clusterInfoInit(g *CommandGetOptions, f util.Factory, clusterInfos map[string]*ClusterInfo) error {
```
Comment (lonelyCZ) [inline pkg/karmadactl/get.go:882]:
```suggestion
func (g *CommandGetOptions) clusterInfoInit(f util.Factory, clusterInfos map[string]*ClusterInfo) error {
```

**Comment by helen-frank** [comment]
test
```shell
 # ./karmadactl --kubeconfig /etc/karmada/karmada-apiserver.config get pod -n test-03                                                                                             [0]
NAME                    CLUSTER            READY   STATUS    RESTARTS   AGE
nginx-f89759699-4m2ml   kind-multi-node1   1/1     Running   0          4d11h
nginx-f89759699-4vvkd   kind-multi-node1   1/1     Running   0          4d11h
# ./karmadactl --kubeconfig /etc/karmada/karmada-apiserver.config get pod -A                                                                                                     [0]
NAMESPACE            NAME                                                CLUSTER            READY   STATUS    RESTARTS   AGE
karmada-system       etcd-0                                              kind-multi-node1   1/1     Running   0          2d14h
karmada-system       karmada-aggregated-apiserver-64c4988754-zjw44       kind-multi-node1   1/1     Running   0          2d14h
karmada-system       karmada-apiserver-58d5fd4888-9b7hc                  kind-multi-node1   1/1     Running   0          2d14h
karmada-system       karmada-controller-manager-795875959b-gjssd         kind-multi-node1   1/1     Running   0          2d14h
karmada-system       karmada-scheduler-6b77dd55f5-4hrtd                  kind-multi-node1   1/1     Running   0          2d14h
karmada-system       karmada-webhook-864bfd745f-gdxss                    kind-multi-node1   1/1     Running   0          2d14h
karmada-system       kube-controller-manager-7f46cd97db-27tdv            kind-multi-node1   1/1     Running   0          2d14h
kube-system          coredns-66bff467f8-6c6lj                            kind-multi-node1   1/1     Running   0          4d12h
kube-system          coredns-66bff467f8-djh2l                            kind-multi-node1   1/1     Running   0          4d12h
kube-system          etcd-multi-node1-control-plane                      kind-multi-node1   1/1     Running   0          4d12h
kube-system          kindnet-259vj                                       kind-multi-node1   1/1     Running   1          4d12h
kube-system          kindnet-2mqxf                                       kind-multi-node1   1/1     Running   0          4d12h
kube-system          kindnet-w2kt4                                       kind-multi-node1   1/1     Running   0          4d12h
kube-system          kube-apiserver-multi-node1-control-plane            kind-multi-node1   1/1     Running   0          4d12h
kube-system          kube-controller-manager-multi-node1-control-plane   kind-multi-node1   1/1     Running   0          4d12h
kube-system          kube-proxy-ccjwz                                    kind-multi-node1   1/1     Running   0          4d12h
kube-system          kube-proxy-fsnb4                                    kind-multi-node1   1/1     Running   0          4d12h
kube-system          kube-proxy-zsvs2                                    kind-multi-node1   1/1     Running   0          4d12h
kube-system          kube-scheduler-multi-node1-control-plane            kind-multi-node1   1/1     Running   0          4d12h
local-path-storage   local-path-provisioner-585bbbbc49-m52bx             kind-multi-node1   1/1     Running   0          4d12h
test-03              nginx-f89759699-4m2ml                               kind-multi-node1   1/1     Running   0          4d11h
test-03              nginx-f89759699-4vvkd                               kind-multi-node1   1/1     Running   0          4d11h
```

File: pkg/karmadactl/get.go [hunk 4]
```diff
@@ -440,7 +441,7 @@ func (g *CommandGetOptions) getObjInfo(wg *sync.WaitGroup, mux *sync.Mutex, f cm
 
 	r := f.NewBuilder().
 		Unstructured().
-		NamespaceParam(g.Namespace).DefaultNamespace().AllNamespaces(g.AllNamespaces).
+		NamespaceParam(*defaultConfigFlags.Namespace).DefaultNamespace().AllNamespaces(g.AllNamespaces).
```
Comment (lonelyCZ) [inline pkg/karmadactl/get.go:444]: There somes problems, the default namespace isn't set.
```
[root@master67 karmada]# ./karmadactl get pod --kubeconfig /etc/karmada/karmada-apiserver.config
NAME                                           CLUSTER    READY   STATUS    RESTARTS       AGE
etcd-0                                         member67   1/1     Running   2              35d
karmada-aggregated-apiserver-b8f7c44d9-2rjz2   member67   1/1     Running   2              35d
karmada-apiserver-74cfd598d8-prfmt             member67   1/1     Running   1              35d
karmada-controller-manager-6b4b75f877-g6mvd    member67   1/1     Running   4              35d
karmada-scheduler-84dfc6f6d4-l42w7             member67   1/1     Running   3              35d
karmada-webhook-9b5c8b94b-b54gq                member67   1/1     Running   2              35d
kube-controller-manager-7f8b7fb754-5mgzm       member67   1/1     Running   4              35d
proxy-server-774dc95c8d-nffdf                  member67   1/1     Running   0              6d21h
coredns-6d56c8448f-6sxdn                       member67   1/1     Running   19             332d
coredns-6d56c8448f-pt9ns                       member67   1/1     Running   19             332d
etcd-master67                                  member67   1/1     Running   16             215d
kube-apiserver-master67                        member67   1/1     Running   10             177d
kube-controller-manager-master67               member67   1/1     Running   26             215d
kube-flannel-ds-klpjg                          member67   1/1     Running   25             332d
kube-proxy-9p86s                               member67   1/1     Running   21             361d
kube-scheduler-master67                        member67   1/1     Running   28             215d
nginx-6799fc88d8-7x6mh                         member68   1/1     Running   0              7d20h
karmada-agent-784fd76b6-7js9c                  member68   1/1     Running   4 (3h5m ago)   20h
proxy-agent-7948957bb7-vkd9t                   member68   1/1     Running   0              20h
coredns-7f6cbbb7b8-77wjk                       member68   1/1     Running   27 (19d ago)   326d
coredns-7f6cbbb7b8-hhp8x                       member68   1/1     Running   26 (19d ago)   326d
etcd-master68                                  member68   1/1     Running   13 (19d ago)   119d
kube-apiserver-master68                        member68   1/1     Running   12 (19d ago)   119d
kube-controller-manager-master68               member68   1/1     Running   17 (19d ago)   119d
kube-flannel-ds-9w82j                          member68   1/1     Running   37 (19d ago)   326d
kube-proxy-v855g                               member68   1/1     Running   17 (19d ago)   166d
kube-scheduler-master68                        member68   1/1     Running   18 (19d ago)   119d
```

v1.3
```
[root@master67 karmada]# karmadactl get pod --kubeconfig /etc/karmada/karmada-apiserver.config
NAME                     CLUSTER    READY   STATUS    RESTARTS   AGE
nginx-6799fc88d8-7x6mh   member68   1/1     Running   0          7d20h

**Comment by helen-frank** [comment]
fixed @lonelyCZ

**Pull request merged**